<div align="right"> Tyler Grimes <br> 
Jan. 21 2016  
</div>
MAC 6106 Homework 2, Ch. 3 #10
<hr>
> ##### 10. Use simulation to approximate $Cov(U,e^U)$, where $U$ is uniform on $(0, 1)$. Compare your approximation with the exact answer.  
  
##### Derivation:  
By definiton of covariance, we have

$$
\begin{aligned}
cov(U, e^U) &= E((U - E(U))(e^U - E(e^U)) \\
&= E(Ue^U) - E(U)E(e^U) \\
\end{aligned}
$$

We know how to approximate the expected values by simulation, so we can use those to approximate the covariance.

##### Algorithm:  
1. Generate $y_i$ from Uniform(0,1).
2. Evaluate $exp(y_i)$ and $yexp(y_i)$.
3. Repeat 1 and 2 for $n = 10^6$ iterations.
4. Evaluate $\frac{\sum_{i=1}^n y_iexp(y_i)}{n} - \frac{\sum_{i=1}^n y_i}{n} \frac{\sum_{i=1}^n exp(y_i)}{n}$.

In [3]:
import math
import random as ran

def simulation(seed, n = 1000000): 
    sum1 = 0; sum2 = 0; sum3 = 0
    ran.seed(seed)
    for i in xrange(n):
        y = ran.uniform(0, 1)
        sum1 = sum1 + y
        sum2 = sum2 + math.exp(y)
        sum3 = sum3 + y*math.exp(y)
    return (sum3)/n - (sum1/n)*(sum2/n)

simulation(seed = 10)

0.14078343923735215

The simulation yields the approximation $cov(U, e^U) \approx 0.1408$

##### Analytical result:  
  
If $U \sim Unif(0, 1)$, then

$$
\begin{aligned}
E(U) &= \int_0^1 u du = \frac{1}{2} \\
E(e^U) &= \int_0^1 e^u du = e - 1 \\
E(Ue^U) &= \int_0^1 ue^u du \\
&= \left[ ue^u \right]_0^1 - \int_0^1 e^u du \\
&= e - (e - 1) \\
&= 1
\end{aligned}
$$

So, we have
$$
Cov(U, e^U) = E(Ue^U) - E(U)E(e^U) = 1 - (e-1) \left( \frac{1}{2} \right) = \frac{3-e}{2} \approx 0.1408591
$$.

The exact result agrees with the simulated result to three decimal places. 